stdlib/Random/test/runtests.jl

```diff
@@ -1250,6 +1250,10 @@ end
     @test length(xs) == 3
 end
 
+@testset "Float32 RNG typo" begin
+    @test length(unique(rand(Float32, 14))) > 10
+end
```

Comment:
Extend to other types as well?  This won't catch similar issues otherwise.

Reply:
The test here is pretty much just to check that the concrete issue is fixed. I don't think there is much value in extending the tests more than that.

Reply:
Well, it'd have caught https://github.com/JuliaLang/julia/pull/55997#issuecomment-2395036394

Reply:
> Why wasn't the test added there?

I was thinking of count iszero tests, but those are brittle at best, especially for Float16 (which generates 2048 unique numbers, so the probability of picking exactly 0 is non-negligible) and especially for small vectors (which was the broken case),  I could hit test failures fairly easily.